### PR TITLE
Fix issues regarding the *empty* sort

### PR DESCRIPTION
**Pull request by stevebeauge**
Fix issues regarding the *empty* sort

 #238
 #237
 #234

The idea is consider a `null` sort instead of an empty one

### CODE REVIEW

**Comment by wobba** [comment]
Please look at my PR for the same at #243 as well. I tested scenarios where you add/remove/change the sortlist and default settings to ensure there's an update happening for all cases. Does your solution work for edits scenarios?

**Comment by stevebeauge** [comment]
> 
> 
> Please look at my PR for the same at #243 as well. I tested scenarios where you add/remove/change the sortlist and default settings to ensure there's an update happening for all cases. Does your solution work for edits scenarios?

Yes. Unless I missed some use cases. I checked the edit scenario, and saw sorting as expected. Of course, a careful test session isn't useless

**Comment by wobba** [comment]
@stevebeauge 

1. Have both sortlist and default sort
2. Change default sort to asc/desc
3. Change sortlist to opposite of desc/asc
4. Remove default sort
5. Verify sortlist is used without refresh

**Comment by stevebeauge** [comment]
My test case works with your repro steps

File: search-parts/src/webparts/searchResults/components/SearchResultsContainer/SearchResultsContainer.tsx [hunk 1]
```diff
@@ -705,20 +705,18 @@ export default class SearchResultsContainer extends React.Component<ISearchResul
...
+    private _getDefaultSortingValues() : Sort {
         if (this.props.sortList.length > 0
             && this.props.sortableFields.length > 0
             && this.props.sortList[0].sortField === this.props.sortableFields[0].sortField) {
-            sortField = this.props.sortList[0].sortField;
-            sortDirection = this.props.sortList[0].sortDirection === ISortFieldDirection.Ascending
-                ? SortDirection.Ascending : SortDirection.Descending;
+            return {
+              Property : this.props.sortList[0].sortField,
+              Direction : this.props.sortList[0].sortDirection === ISortFieldDirection.Ascending
```
Comment (wobba) [inline search-parts/src/webparts/searchResults/components/SearchResultsContainer/SearchResultsContainer.tsx:714]: Should be `Direction : this.props.sortableFields[0].sortDirection === ISortFieldDirection.Ascending` I believe.

**Comment by wobba** [comment]
Let me pull your branch and test as well.... less changes than mine in a way.

**Comment by wobba** [comment]
This fails:
1. Add webpart
2. Change edit sort order, set Created to descending. No update on the page.
3. Add a default sort prop, no update/change

I'm inclined to merge my code ;)

**Pull request closed without merging**